libskale/State.cpp

```diff
@@ -817,7 +824,19 @@ std::pair< ExecutionResult, TransactionReceipt > State::execute( EnvInfo const&
 
         removeEmptyAccounts = _envInfo.number() >= _sealEngine.chainParams().EIP158ForkBlock;
         commit( removeEmptyAccounts ? State::CommitBehaviour::RemoveEmptyAccounts :
-                                      State::CommitBehaviour::KeepEmptyAccounts );
+                                      State::CommitBehaviour::KeepEmptyAccounts,
+            [&]( std::shared_ptr< dev::db::DatabaseFace > /*db*/,
+                std::unique_ptr< dev::db::WriteBatchFace >& writeBatch ) {
+                if ( isSaveLastTxHash ) {
```

Comment:
Why this cannot be always true?

Reply:
I am just providing new and old versions of methods at once
isSaveLastTxHash = true -> my new version needed for partial catch up

Reply:
So can we just remove this parameter and work as if it's true?

Reply:
yes, right, but I need it initialized

Reply:
State::execute() is invoked several times in code and I am sure I need new isSaveLastTxHash parameter to be true only in one place, and not sure about other places, so isSaveLastTxHash is too early to remove